src/database.h

```diff
@@ -161,11 +159,8 @@ class DBResult
 			}
 
 			T data;
-			try {
-				data = boost::lexical_cast<T>(row[it->second]);
-			} catch (boost::bad_lexical_cast&) {
-				data = 0;
-			}
+			std::stringstream ss(row[it->second]);
+			ss >> data;
```

Comment:
For DBResult the performance should probably be measured to see if it's a significant difference (which I think it is), and if it is, we should add specialized templates for the most common types.